src/sentry/auth/superuser.py

```diff
@@ -304,17 +305,20 @@ def set_logged_in(self, user, current_datetime=None):
                 },
             )
 
-        self._set_logged_in(
-            expires=current_datetime + MAX_AGE,
-            token=token,
-            user=user,
-            current_datetime=current_datetime,
-        )
+            self._set_logged_in(
+                expires=current_datetime + MAX_AGE,
+                token=token,
+                user=user,
+                current_datetime=current_datetime,
+            )
 
-        logger.info(
-            "superuser.logged-in",
-            extra={"ip_address": request.META["REMOTE_ADDR"], "user_id": user.id},
-        )
+            logger.info(
+                "superuser.logged-in",
+                extra={"ip_address": request.META["REMOTE_ADDR"], "user_id": user.id},
+            )
+
+        except AttributeError:
```

Comment:
What is this anticipating? `AttributeError` is quite broad.

If this is meant to proceed even if one of the `logger.info` calls fails, that's good, but as written it would cause us to skip calling `_set_logged_in` if the first `logger.info` chokes.

Reply:
This is anticipating if there are any missing information in request.user. If that happens we want to skip `_set_logged_in` and the subsequent  `superuser.logged-in` log

Reply:
@maxiuyuan how would that ever happen?

Reply:
is there a test case that covers this?